databuilder/backends/base.py

```diff
@@ -39,67 +39,51 @@ def get_table_expression(self, table_name, schema):
         Raises:
             ValueError: If unknown table passed in
         """
-        # TODO: We currently ignore the `schema` argument here. But I think we should
-        # move towards having the supplied schema define the column types or, if not
-        # that, then we should validate that the types match.
-        return self.tables[table_name].get_expression(table_name)
+        return self.tables[table_name].get_expression(table_name, schema)
 
 
 class SQLTable:
     def learn_patient_join(self, source):
-        raise NotImplementedError()
-
-    def _make_columns(self):
-        return [
-            self._make_column(name, column) for name, column in self.columns.items()
-        ]
-
-    def _make_column(self, name, column):
-        source = column.source or name
-        type_ = TYPES_BY_NAME[column.type].value
-        sql_column = sqlalchemy.Column(source, type_, key=name)
-        return sql_column
+        self.patient_join_column = source
```

Comment:
Previously it was a bit odd that `QueryTable.learn_patient_join()` ignored its argument and triggered something that could have happened in the constructor. Now it's even more weird that this method on the parent class sets a field that is only needed by one subclass.

I'm pretty sure that there is a cleaner way of arranging what needs to happen here, but for the life of me I can't see what it is.

Reply:
Yeah, it's all a bit round the houses isn't it? I didn't want to spend too much time reworking this stuff here, beyond what was needed to get the API surface right. There may be something we could do with descriptors here to make it all drop out more cleanly.